src/roomHandlers/fishingRoomHandler.ts

```diff
@@ -0,0 +1,114 @@
+import { RoomHandler } from './roomHandler';
+
+import {
+  Command,
+  CommandHandlersByCommandByPlayerStates,
+  DEFAULT_COMMAND,
+  NarrowedRoomState,
+  NullableCommandHandler,
+  StateDescriptionAccessor,
+} from '../utils/types';
+
+const roomType = 'fishingRoom';
+type TRoomType = typeof roomType;
+type TRoomState = NarrowedRoomState<TRoomType>;
+type TStateDescriptionAccessor = StateDescriptionAccessor<TRoomState>;
+type TCommandHandlersByCommandByPlayerStates =
+  CommandHandlersByCommandByPlayerStates<TRoomType>;
+type TNullableCommandHandler = NullableCommandHandler<TRoomType>;
+
+const stateDescriptionAccessor: TStateDescriptionAccessor = {
+  AtEntrance: ({ fishCaught, isRodBroken }) => {
+    const poleText = isRodBroken ? 'broken' : 'working';
+
+    return {
+      playerStateDescription: `You are on a dock with a ${poleText} fishing pole. You have caught ${fishCaught} fish so far!`,
+      availableCommands: ['fish', 'fix', 'leave'],
+    };
+  },
+  Fishing: {
+    playerStateDescription: 'Hopefully the fish are hungry today.',
+    availableCommands: ['stop', 'fish'],
+  },
+};
+
+const commandHandlersByCommandByPlayerState: TCommandHandlersByCommandByPlayerStates =
+  {
+    AtEntrance: {
+      fish: (roomState) => ({
+        commandDescription: `You sit down and start to get ready to fish. ${
+          roomState.isRodBroken
+            ? 'But your rod is broken and you have to fix it.'
+            : 'This is going to be alot of fun!'
+        }`,
+        roomState: {
+          ...roomState,
+          playerState: roomState.isRodBroken ? 'AtEntrance' : 'Fishing',
+        },
+      }),
+      fix: (roomState) => ({
+        commandDescription: 'You fix your fishing line.',
+        roomState: {
+          ...roomState,
+          playerState: 'AtEntrance',
+          isRodBroken: false,
+        },
+      }),
+      leave: () => ({
+        commandDescription: 'You say goodbye to the fish and leave.',
+        roomState: null,
+      }),
+    },
+    Fishing: {
+      stop: (roomState) => ({
+        commandDescription: 'You stop fishing.',
+        roomState: {
+          ...roomState,
+          playerState: 'AtEntrance',
+        },
+      }),
+      fish: (roomState) => ({
+        commandDescription: `You cast your fishing line. ${
+          roomState.randomNumber === 0
+            ? 'And nothing happen, maybe try again?'
+            : 'And something happen!'
+        }`,
+        roomState: {
+          ...roomState,
+          playerState: roomState.randomNumber === 0 ? 'Fishing' : 'AtEntrance',
+          fishCaught:
+            roomState.randomNumber === 1
+              ? roomState.fishCaught + 1
+              : roomState.fishCaught,
+          isRodBroken: roomState.randomNumber === 2,
+          randomNumber: Math.floor(Math.random() * 3),
+        },
```

Comment:
3. It can be generated in the the function, but I'm not sure how to test that without stubbing something. I could update the [run](https://github.com/normal-devs/klicker-knight/blob/main/src/roomHandlers/roomHandler.ts#L32-L46) to take a third parameter "options", and pass that into `commandHandler`.

Reply:
1. Yes, that should be `continue`. Updated to match the docs.
2.  I was wondering about that... I thought the user could figure what happen if they went back to the `AtEntrance` state. But I like the idea of the command description outputting more info. Updated `command description`.

Reply:
I'm ok with stubbing Math.random for now. I think if we are going to inject context anywhere it should be in the fishing room handler constructor, but IMO that's overly complex for this PR.

I'm making a PR to allow you to stub Math.random in your PR.

Reply:
Allow stubbing Math.random https://github.com/normal-devs/klicker-knight/pull/45